extensions/cactus-plugin-htlc-coordinator/src/test/typescript/integration/plugin-htlc-coordinator/withdraw-counterparty-endpoint.test.ts

```diff
@@ -0,0 +1,295 @@
+    contractName: TestTokenJSON.contractName,
+    contractAbi: TestTokenJSON.abi,
+    bytecode: TestTokenJSON.bytecode,
+    web3SigningCredential,
+    keychainId,
+    constructorArgs: ["100", "token", "2", "TKN"],
+    gas: estimatedGas,
+  });
+  t.ok(deployOutToken, "deployContract() output is truthy OK");
+  t.ok(
+    deployOutToken.transactionReceipt,
+    "deployContract() output.transactionReceipt is truthy OK",
+  );
+  t.ok(
+    deployOutToken.transactionReceipt.contractAddress,
+    "deployContract() output.transactionReceipt.contractAddress is truthy OK",
+  );
+  const tokenAddress = deployOutToken.transactionReceipt
+    .contractAddress as string;
+
+  t.comment("Approve 10 Tokens to HashTimeLockAddress");
+  const { success } = await connector.invokeContract({
+    contractName: TestTokenJSON.contractName,
+    keychainId,
+    signingCredential: web3SigningCredential,
+    invocationType: EthContractInvocationType.Send,
+    methodName: "approve",
+    params: [contractAddress, "10"],
+    gas: estimatedGas,
+  });
+  t.equal(success, true, "approve() transactionReceipt.status is true OK");
+
+  t.comment("Get account balance");
+  const responseBalance = await connector.invokeContract({
+    contractName: TestTokenJSON.contractName,
+    keychainId,
+    signingCredential: web3SigningCredential,
+    invocationType: EthContractInvocationType.Call,
+    methodName: "balanceOf",
+    params: [firstHighNetWorthAccount],
+  });
+  t.equal(responseBalance.callOutput, "100", "balance of account is 100 OK");
+
+  t.comment("Get HashTimeLock contract and account allowance");
+  const { callOutput } = await connector.invokeContract({
+    contractName: TestTokenJSON.contractName,
+    keychainId,
+    signingCredential: web3SigningCredential,
+    invocationType: EthContractInvocationType.Call,
+    methodName: "allowance",
+    params: [firstHighNetWorthAccount, contractAddress],
+  });
+  t.equal(callOutput, "10", "callOutput() is 10 OK");
+
+  t.comment("Create and initialize own HTLC");
+  const ownHTLCRequest: OwnHTLCRequest = {
+    htlcPackage: HtlcPackage.BesuErc20,
+    connectorId,
+    keychainId,
+    constructorArgs: [],
+    web3SigningCredential,
+    inputAmount: 10,
+    outputAmount: 1,
+    expiration,
+    hashLock,
+    tokenAddress,
+    receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    gas: estimatedGas,
+  };
+
+  const response = await api.apiV1OwnCoordinator(ownHTLCRequest);
+  t.equal(response.status, 200, "response status is 200 OK");
+  t.equal(response.data.success, true, "response sucess is true");
+  t.ok(
+    response.data,
+    "pluginHTLCCoordinator.apiV1OwnCoordinator() output is truthy OK",
+  );
+  t.ok(
+    response.data.out.transactionReceipt,
+    "pluginHTLCCoordinator.apiV1OwnCoordinator() output.transactionReceipt is truthy OK",
+  );
+
+  t.comment("Get HTLC id");
+  const responseTxId = await connector.invokeContract({
```

Comment:
@jagpreetsinghsasan Please assert more specifically the status code being 2xx and also refactor this to use the API client not the connector instance directly.